lisa/target.py

```diff
@@ -664,6 +665,68 @@ def install_tools(self, tools):
             self.target.install(tool)
             self._installed_tools.add(tool)
 
+    def build_kmodule(self, kmodule):
+        """
+        Compile kernel module that we want to load at the target at some point
+
+        :param kmodule: The name of the module to build
+        :type kmodule: str
+        """
+        logger = self.get_logger()
+
+        cmd = '{}/tools/kmodules/build_module'.format(LISA_HOME)
+
+        logger.info('Building kernel module [{}] using [{}] kernel tree'.format(kmodule, self._kernel_src))
+
+        failed = subprocess.call([cmd, kmodule, self._kernel_src])
```

Comment:
I don't get you. I did test this to verify that we detect we get the right return code if we fail, which in this case we just log and carry on.

Are you saying better to use check_call() to enforce raising an error?

Reply:
Yes, I'd rather have this fail if there is any issue with the module in one way or another. Since that feature has to stay optional in any case, if it breaks with an exception, the user has the choice of not trying to use that module to "fix" the issue. The caller of that code can therefore expect this function to always do its job if it returns (return != raise), which is the usual way of proceeding in Python.